internal/statemachine/classic_states.go

```diff
@@ -658,88 +658,78 @@ func (stateMachine *StateMachine) preseedClassicImage() error {
-		}
+	src := stateMachine.tempDirs.chroot
 
-		for _, srcFile := range files {
-			srcFile := filepath.Join(src, srcFile.Name())
-			if err := osutilCopySpecialFile(srcFile, classicStateMachine.tempDirs.rootfs); err != nil {
-				return fmt.Errorf("Error copying rootfs: %s", err.Error())
-			}
-		}
+	files, err := ioutilReadDir(src)
+	if err != nil {
+		return fmt.Errorf("Error reading unpack/chroot dir: %s", err.Error())
+	}
 
-		fstabPath := filepath.Join(classicStateMachine.tempDirs.rootfs, "etc", "fstab")
-		fstabBytes, err := ioutilReadFile(fstabPath)
-		if err == nil {
-			if !strings.Contains(string(fstabBytes), "LABEL=writable") {
-				re := regexp.MustCompile(`(?m:^LABEL=\S+\s+/\s+(.*)$)`)
-				newContents := re.ReplaceAll(fstabBytes, []byte("LABEL=writable\t/\t$1"))
-				if !strings.Contains(string(newContents), "LABEL=writable") {
-					newContents = []byte("LABEL=writable   /    ext4   defaults    0 0")
-				}
-				err := ioutilWriteFile(fstabPath, newContents, 0644)
-				if err != nil {
-					return fmt.Errorf("Error writing to fstab: %s", err.Error())
-				}
-			}
+	for _, srcFile := range files {
+		srcFile := filepath.Join(src, srcFile.Name())
+		if err := osutilCopySpecialFile(srcFile, classicStateMachine.tempDirs.rootfs); err != nil {
+			return fmt.Errorf("Error copying rootfs: %s", err.Error())
 		}
+	}
 
-		if classicStateMachine.commonFlags.CloudInit != "" {
-			seedDir := filepath.Join(classicStateMachine.tempDirs.rootfs, "var", "lib", "cloud", "seed")
-			cloudDir := filepath.Join(seedDir, "nocloud-net")
-			err := osMkdirAll(cloudDir, 0756)
-			if err != nil && !os.IsExist(err) {
-				return fmt.Errorf("Error creating cloud-init dir: %s", err.Error())
+	fstabPath := filepath.Join(classicStateMachine.tempDirs.rootfs, "etc", "fstab")
+	fstabBytes, err := ioutilReadFile(fstabPath)
+	if err == nil {
+		if !strings.Contains(string(fstabBytes), "LABEL=writable") {
+			re := regexp.MustCompile(`(?m:^LABEL=\S+\s+/\s+(.*)$)`)
+			newContents := re.ReplaceAll(fstabBytes, []byte("LABEL=writable\t/\t$1"))
+			if !strings.Contains(string(newContents), "LABEL=writable") {
+				newContents = []byte("LABEL=writable   /    ext4   defaults    0 0")
 			}
-			metadataFile := filepath.Join(cloudDir, "meta-data")
-			metadataIO, err := osOpenFile(metadataFile, os.O_CREATE|os.O_WRONLY, 0644)
+			err := ioutilWriteFile(fstabPath, newContents, 0644)
 			if err != nil {
-				return fmt.Errorf("Error opening cloud-init meta-data file: %s", err.Error())
+				return fmt.Errorf("Error writing to fstab: %s", err.Error())
 			}
-			metadataIO.Write([]byte("instance-id: nocloud-static"))
-			metadataIO.Close()
+		}
+	}
 
-			userdataFile := filepath.Join(cloudDir, "user-data")
-			err = osutilCopyFile(classicStateMachine.commonFlags.CloudInit,
-				userdataFile, osutil.CopyFlagDefault)
-			if err != nil {
-				return fmt.Errorf("Error copying cloud-init: %s", err.Error())
-			}
-		}*/
+	if classicStateMachine.commonFlags.CloudInit != "" {
```

Comment:
This feels fine for instance. I know we're working on having a cloud-init configuration via the image-definition file, but I think keeping the old bits around for overriding the image-definition configuration makes sense.